File: pinyougou-parent/pinyougou-user-web/src/main/java/com/pinyougou/user/controller/UserController.java
package com.pinyougou.user.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbUser;
import com.pinyougou.user.service.UserService;
import entity.Result;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import utils.PhoneFormatCheckUtils;

@RestController
@RequestMapping("/user")
public class UserController {

    @Reference
    private UserService userService;

    @RequestMapping("/register")
    public Result register(@RequestBody TbUser user, String verifyCode) throws Exception {
        try {
            //校验验证码
            if (!userService.checkVerifyCode(user.getPhone(), verifyCode)) {
                return new Result(false, "验证码输入错误!");
            }
            userService.registerUser(user);
            return new Result(true, "注册成功!");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "注册失败!");
        }
    }

    @RequestMapping("/sendVerifyCode")
    public Result sendVerifyCode(String phone) throws Exception {
        //校验验手机号码
        if (!PhoneFormatCheckUtils.isChinaPhoneLegal(phone)) {
            return new Result(false, "手机号格式不正确!");
        }
        try {
            userService.createVerifyCode(phone);
            return new Result(true, "验证码发送成功!");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "验证码发送失败!");
        }
    }
}

File: pinyougou-parent/pinyougou-user-web/src/main/webapp/plugins/angularjs/controller/userController.js
//控制器
app.controller('userController', function ($scope, $controller, userService) {

    //继承baseController
    $controller('baseController', {$scope: $scope});

    $scope.register = function () {
        if ($scope.entity.password != $scope.password || $scope.password == '') {
            alert("两次密码输入不一致,请重新输入!");
            $scope.password = '';
            $scope.entity.password = '';
            return;
        }
        userService.register($scope.entity, $scope.verifyCode).success(function (data) {
            alert(data.message);
        });
    };

    $scope.sendVerifyCode = function () {
        if ($scope.entity.phone == null || $scope.entity.phone == '') {
            alert("请输入手机号!");
            return;
        }
        userService.sendVerifyCode($scope.entity.phone).success(function (data) {
            alert(data.message);
        });
    }
});